### PR TITLE
network(dm): metering network usage from lightning and dumpling

### LINKED ISSUE #7933: DM: network metering from dumpling and lightning

### PR DESCRIPTION
**Pull request by buchuitoudegou**
### What problem does this PR solve?

Issue Number: close #7933 

### What is changed and how it works?
- move TCPConnWithCounter to TiDB
- pass io counter pointer to dumpling/lightning

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

File: dm/dumpling/dumpling.go [hunk 1]
```diff
@@ -74,6 +85,10 @@ func (m *Dumpling) Init(ctx context.Context) error {
 	if m.dumpConfig, err = m.constructArgs(ctx); err != nil {
 		return err
 	}
+	if m.cfg.DumpIOTotalBytes != nil {
+		m.dumpConfig.IOTotalBytes = m.cfg.DumpIOTotalBytes
```
[the remark this answers is not in the document]
Comment (buchuitoudegou) [reply dm/dumpling/dumpling.go:89]: No. In dumpling, this field is called `IOTotalBytes`. In dm, we have two fields to meter the network, so I put a prefix to separate them.

**Review by lance6716 — APPROVED** [review]
do we need to meter syncer's binlog reading connection?

**Comment by buchuitoudegou** [comment]
> do we need to meter syncer's binlog reading connection?

Yes. The network consumption is from the upstream, which is probably connected by private link or vpc peering.

**Review by D3Hunter — APPROVED** [review]

**Comment by okJiang** [comment]
> /hold
> 
> impl in another pr #7954

should this pr close? @buchuitoudegou

**Pull request closed without merging**